utilities/validations.py

```diff
@@ -129,13 +129,29 @@ def validate_room_labels(**kwargs):
 
 def validate_structure_id(**kwargs):
     """
-    Function to validate that a structure id exists in structures
+    Function to validate that a structure id exists or is unique in structures
     table
     :param structure_id
     """
-    structure_id = kwargs.get('structure_id')
-    structure = StructureModel.query.filter_by(
-        structure_id=structure_id
-    ).first()
-    if structure is None:
-        raise GraphQLError("Structure id does not exist")
+
+    if kwargs.get('structure_id'):
+        structure = StructureModel.query.filter_by(
+            structure_id=kwargs.get('structure_id')
+        ).first()
+        if structure is None:
+            raise GraphQLError("Structure id does not exist")
+    else:
+        structure_id_list = [structure['structure_id']
+                             for structure in kwargs['data']]
+        duplicate_structures = []
+        for structure_id in structure_id_list:
+            if structure_id_list.count(structure_id) > 1:
```

Comment:
Why don't you just return the error message if this condition is true, instead of appending the ids in the list and checking if the list has ids?

Reply:
Also, check `[[Finishes #165904360]` in your commit message

Reply:
Thanks. Updated.